firedrake/functionspaceimpl.py

```diff
@@ -321,6 +321,14 @@ def __iter__(self):
         return iter(self.subfunctions)
 
     def __getitem__(self, i):
+        from firedrake.functionspace import MixedFunctionSpace
+        if isinstance(i, list):
```

Comment:
We should allow any iterable, not just a list.
```suggestion
        if not isinstance(i, int):
```

Reply:
I think the behaviour of `__getitem__` is a bit subtle here. Calling `obj[1, 2, 3]` is *exactly the same* as calling `obj[(1, 2, 3)]` which is not quite the same as `obj[[1, 2, 3]]`.

I think having something like `space[1, 2]` would suggest to someone that `space` is somehow nested, whereas `space[[1, 2]]` suggests a subset of the indices.

Reply:
No longer relevant